### PR TITLE
adding initial generator cli

### PR DESCRIPTION
**Pull request by dskatz**
Adding the initial CLI structure using cobra.  Generators are stubbed and will be filled in a future PR.

### CODE REVIEW

File: internal/cli/flags.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package cli
+
+import (
+	"github.com/spf13/pflag"
+	gengoargs "k8s.io/gengo/args"
+)
+
+func flagLogLevel(flags *pflag.FlagSet) {
+	flags.String("log-level", "info", "Configure log level")
+}
```
Comment (yuzhouliu9) [inline internal/cli/flags.go:10]: I think this function can be removed and instead included in `flagGeneratorArgs`?

Comment (dskatz) [reply internal/cli/flags.go:10]: `flagGeneratorArgs` is specifically for setting up flags that affect the generator itself.

File: internal/cli/root.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+	cmd := &cobra.Command{
+		Use:   "kanopy-codegen",
+		Short: "Kanopy Builder code generator",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+
+			if err := viper.BindPFlags(cmd.Flags()); err != nil {
+				return err
+			}
+
```
[the remark this answers is not in the document]
Comment (dskatz) [reply internal/cli/root.go:40]: Yeah, completely forgot about that.

File: internal/cli/root.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			return nil
+		},
+		RunE: rootCommand.runE,
+	}
+
+	flags := cmd.PersistentFlags()
+	customArgs := &generators.CustomArgs{}
+	flagLogLevel(flags)
+	flags.StringSliceVar(&customArgs.BoundingDirs, "bounding-dirs", customArgs.BoundingDirs, "specify directories to bound the generation")
```
[the remark this answers is not in the document]
Comment (dskatz) [reply internal/cli/root.go]: I created a separation between the "default flags" and the custom ones.

File: internal/cli/root.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+func NewRootCommand(opts ...func(*rootCommand)) *cobra.Command {
+	rootCommand := &rootCommand{
+		GeneratorArgs: gengoargs.Default(),
+	}
+
+	for _, opt := range opts {
+		opt(rootCommand)
+	}
+
+	rootCommand.GeneratorArgs.OutputFileBaseName = "zz_generated_builders"
```
[the remark this answers is not in the document]
Comment (dskatz) [reply internal/cli/root.go:30]: Unfortunately, the type of `rootCommand.GeneratorArgs.CustomArgs` is an `interface{}`.   A local variable of the `generators.CustomArgs` type must be used to bind to the flag.  Not using viper here felt more natural than having to translate from that.

[the remark this answers is not in the document]
Comment (yuzhouliu9) [reply internal/cli/root.go:30]: Oh gotcha

**Review by yuzhouliu9 — APPROVED** [review]

**Pull request merged**